pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -482,7 +482,10 @@ def setup_main(solve_data, config, fp, regularization_problem):
                 solve_data.working_model,
                 discrete_only=config.fp_discrete_only)
     elif regularization_problem:
-        if MindtPy.objective_list[0].expr.polynomial_degree() in {1, 0}:
+        # The epigraph constraint is very "flat" for branching rules.
+        # In ROA, if the objective function is linear, the original objective function is used in the MIP problem. 
+        # In the MIP projection problem, we need to reactivate the epigraph constraint(objective_constr).
+        if MindtPy.objective_list[0].expr.polynomial_degree() in solve_data.mip_objective_polynomial_degree:
```

Comment:
This would now work even if the objective is quadratic and the new option is active right? If so, update comment above

Reply:
Yes.